src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -259,7 +259,22 @@ void InterpreterCreateUserQuery::updateUserFromQuery(User & user, const ASTCreat
     if (query.auth_data)
         auth_data = AuthenticationData::fromAST(*query.auth_data, {}, !query.attach);
 
-    updateUserFromQueryImpl(user, query, auth_data, {}, {}, {}, {}, {}, allow_no_password, allow_plaintext_password, true);
+    std::optional<time_t> valid_until;
```

Comment:
Doesn't it make more sense to do this inside `updateUserFromQueryImpl` instead of duplicating the code. We always pass the query itself, so passing the parsed value doesn't make much sense unless it was changed elsewhere

Reply:
It is not exactly the same since we parse the date time in a specific (more strict) format in case of an attach query and don't fill in the substitutions. `updateUserFromQueryImpl` is basically a setter function that is passed to access storage, so I am not sure if we want to throw parsing or other exceptions caused by bad arguments passed by user from there.

Otherwise, I can do it if you think it makes sense.

Reply:
I implemented an intermediate option with parsing taking place in a separate function, let me know if you think I should move it into `updateUserFromQueryImpl`